tests/test_xtai_calendar.py

```diff
@@ -0,0 +1,117 @@
+from unittest import TestCase
+import pandas as pd
+from pytz import UTC
+
+from .test_trading_calendar import NoDSTExchangeCalendarTestBase
+from trading_calendars.exchange_calendar_xtai import XTAIExchangeCalendar
+
+
+class XTAICalendarTestCase(NoDSTExchangeCalendarTestBase, TestCase):
+
+    answer_key_filename = 'xtai'
+    calendar_class = XTAIExchangeCalendar
+
+    # The XTAI is open from 9:00AM to 1:30PM
+    MAX_SESSION_HOURS = 4.5
+
+    HAVE_EARLY_CLOSES = False
+
+    # A holiday falls on one of the default dates for these test
+    # values, so overwrite here.
+    MINUTE_INDEX_TO_SESSION_LABELS_START = pd.Timestamp('2011-01-06')
+    MINUTE_INDEX_TO_SESSION_LABELS_END = pd.Timestamp('2011-04-06')
+
+    def test_regular_holidays(self):
+        all_sessions = self.calendar.all_sessions
+
+        expected_holidays = [
+            pd.Timestamp('2019-01-01', tz=UTC),  # New Year's Day
+            pd.Timestamp('2019-02-04', tz=UTC),  # Chinese New Year's Eve
+            pd.Timestamp('2019-02-05', tz=UTC),  # Chinese New Year
+            pd.Timestamp('2019-02-28', tz=UTC),  # Peace Memorial Day
+            pd.Timestamp('2019-04-04', tz=UTC),  # Women and Children's Day
+            pd.Timestamp('2019-04-05', tz=UTC),  # Tomb Sweeping Day
+            pd.Timestamp('2019-05-01', tz=UTC),  # Labour Day
+            pd.Timestamp('2019-06-07', tz=UTC),  # Dragon Boat Festival
+            pd.Timestamp('2019-09-13', tz=UTC),  # Mid-Autumn Festival
+            pd.Timestamp('2019-10-10', tz=UTC),  # National Day
+        ]
+
+        for holiday_label in expected_holidays:
+            self.assertNotIn(holiday_label, all_sessions)
+
+    def test_holidays_fall_on_weekend(self):
+        all_sessions = self.calendar.all_sessions
+
+        # Holidays falling on weekends are made up, so verify appropriate
+        # Friday/Monday is a holiday.
+        expected_holidays = [
+            # New Year's Day on Sunday, Jan 1st.
+            pd.Timestamp('2017-01-02', tz=UTC),
+            # Chinese New Year on Saturday, Jan 28th.
```

Comment:
I agree, but it might be a bit messy because Taiwan observes a bunch of holidays for Chinese New Year, and generally it wasn't completely clear how they went about choosing those days (i.e., how many days are we taking off before Chinese New Year and how many days are we taking off after).  Because of this, I basically just took a chunk of dates that are always observed as holidays (I think two days before and two days after), and then hard-coded any extra holidays that would occasionally appear either before or after that block.  So what this means is that in some (but not all) years there could be a Chinese New Year's Eve-Eve-Eve.